ratelimits/limit.go

```diff
@@ -189,16 +214,21 @@ func loadAndParseOverrideLimits(path string) (limits, error) {
 	return parsed, nil
 }
 
-// loadAndParseDefaultLimits loads default limits from YAML, validates them, and
-// parses them into a map of limits keyed by 'Name'.
+// loadAndParseDefaultLimits loads and parses default limits from a YAML file.
 func loadAndParseDefaultLimits(path string) (limits, error) {
 	fromFile, err := loadDefaults(path)
 	if err != nil {
 		return nil, err
 	}
-	parsed := make(limits, len(fromFile))
 
-	for k, v := range fromFile {
+	return parseDefaultLimits(fromFile)
+}
+
+// parseDefaultLimits validates a map of default limits and rekeys it by 'Name'.
+func parseDefaultLimits(newDefaultLimits limits) (limits, error) {
+	parsed := make(limits, len(newDefaultLimits))
```

Comment:
Given that we can't (and therefore don't) do this same optimization in `parseDefaultLimits`, I think it's probably not worth doing it here either. This code gets run once at startup, I'm not worried about how many times it resizes the underlying map.
```suggestion
parsed := make(limits)
```